File: app/src/main/java/edu/gatech/spellastory/domain/Word.java
package edu.gatech.spellastory.domain;

import java.io.Serializable;
import java.util.List;

public class Word implements Serializable {

    private String spelling;
    private List<Phoneme> phonemes;
    private boolean complete;

    public Word(String spelling, List<Phoneme> phonemes, boolean complete) {
        this.spelling = spelling;
        this.phonemes = phonemes;
        this.complete = complete;
    }

    public String getSpelling() {
        return spelling;
    }

    public List<Phoneme> getPhonemes() {
        return phonemes;
    }

    public boolean isComplete() {
        return complete;
    }

    public void setComplete(boolean complete) {
        this.complete = complete;
    }

    @Override
    public String toString() {
        return "Word{" +
                "spelling='" + spelling + '\'' +
                ", phonemes=" + phonemes +
                '}';
    }
}

File: app/src/main/java/edu/gatech/spellastory/domain/Word.kt
package edu.gatech.spellastory.domain

import java.io.Serializable

class Word(val spelling: String, val phonemes: List<Phoneme>, val category: Category?) : Serializable {
    fun contains(code: String): Boolean = code in phonemes.map { it.code }
    fun validate() {
        val phonemeSpelling = phonemes.joinToString("") { it.spelling }
        if (!spelling.equals(phonemeSpelling, true))
            throw Exception("$spelling doesn't match its phonemes $phonemeSpelling")
    }
}